crates/common/account_manager/src/keystore.rs

```diff
@@ -0,0 +1,327 @@
+    }
+
+    /// Create a new keystore from seed phrase and key parameters
+    pub fn from_seed_phrase(
+        seed_phrase: &str,
+        lifetime: u32,
+        activation_epoch: u32,
+        description: Option<String>,
+        path: Option<String>,
+    ) -> Self {
+        let uuid = Uuid::new_v4();
+
+        // Generate random salt for KDF (32 bytes)
+        let salt = hex::encode(rand::random::<[u8; 32]>());
+
+        // Generate random nonce for AES-GCM (12 bytes)
+        let nonce = hex::encode(rand::random::<[u8; 12]>());
+
+        // Generate random tag for AES-GCM (16 bytes)
+        let tag = hex::encode(rand::random::<[u8; 16]>());
+
+        // Store the seed phrase as encrypted data (hex encoded)
+        let ciphertext = hex::encode(seed_phrase.as_bytes());
+
+        let kdf = KdfParams::new_full(65536, 3, 4, salt);
+        let cipher = CipherParams::new(nonce, tag, ciphertext);
+        let crypto = CryptoParams { kdf, cipher };
+        let keytype = KeyType::new(lifetime, activation_epoch);
+
+        let mut keystore = Self::new(crypto, keytype, uuid);
+        keystore.description = description;
+        keystore.path = path;
+
+        keystore
+    }
+
+    /// Validate the keystore structure
+    pub fn validate(&self) -> Result<(), String> {
+        // Check version
+        if self.version != 5 {
+            return Err("Version must be 5".to_string());
+        }
+
+        // Check quantum_secure flag
+        if !self.quantum_secure {
+            return Err("quantum_secure must be true".to_string());
+        }
+
+        // Check KDF function
+        if self.crypto.kdf.function != "argon2id" {
+            return Err("KDF function must be argon2id".to_string());
+        }
+
+        // Check cipher function
+        if self.crypto.cipher.function != "aes-256-gcm" {
+            return Err("Cipher function must be aes-256-gcm".to_string());
+        }
+
+        // Check keytype function
+        if self.keytype.function != "xmss-poisedon2-ots-seed" {
+            return Err("Keytype function must be xmss-poisedon2-ots-seed".to_string());
+        }
+
+        // Validate hex strings
+        self.validate_hex_string(&self.crypto.cipher.ciphertext, "ciphertext")?;
+        self.validate_hex_string(&self.crypto.cipher.params.nonce, "nonce")?;
+        self.validate_hex_string(&self.crypto.cipher.params.tag, "tag")?;
+
+        // Validate salt
+        match &self.crypto.kdf.params {
+            KdfParamsInner::Full { salt, .. } | KdfParamsInner::Short { salt, .. } => {
+                self.validate_hex_string(salt, "salt")?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Helper to validate hex strings
+    fn validate_hex_string(&self, hex_str: &str, field_name: &str) -> Result<(), String> {
+        if hex_str.chars().all(|c| c.is_ascii_hexdigit()) {
+            Ok(())
+        } else {
+            Err(format!("{} must be a valid hex string", field_name))
+        }
+    }
```

Comment:
```suggestion
    /// Helper to validate hex strings
    fn validate_hex_string(&self, hex_str: &str, field_name: &str) -> anyhow::Result<()> {
        ensure!(hex_str.chars().all(|c| c.is_ascii_hexdigit(), format!("{field_name} must be a valid hex string"));
        Ok(())
    }
```

please always inline variables in strings, also do Jun's suggestion, this doesn't have any logic unique to the keystore